misk/src/test/kotlin/misk/concurrent/RealExecutorServiceFactoryTest.kt

```diff
@@ -90,4 +90,20 @@ internal class RealExecutorServiceFactoryTest {
     // ...let the executor service finally shut down.
     latch.countDown()
   }
+
+  @Test fun placeholderRequiredButAbsent() {
+    val factory = RealExecutorServiceFactory(FakeClock())
+    val exception = assertFailsWith<IllegalStateException> {
+      factory.unbounded("no-placeholder")
+    }
+    assertThat(exception).hasMessage("thread index %d placeholder missing: no-placeholder")
+  }
+
+  @Test fun placeholderForbiddenButPresent() {
+    val factory = RealExecutorServiceFactory(FakeClock())
+    val exception = assertFailsWith<IllegalStateException> {
+      factory.single("unexpected-%d")
+    }
+    assertThat(exception).hasMessage("thread index %d placeholder unexpected: unexpected-%d")
+  }
 }
```

Comment:
I think you want all the various fixed cases.

At least one for the failure of `>1` and no `%d`